### ISSUE TITLE
"board listall" and "core search --additional-urls" don't seem to work after a fresh install via homebrew

### ISSUE THREAD

**Issue by alranel** [arduino/arduino-cli#546]
### Current behavior

When no cores are installed, I would expect that `listall` returned all the _existing_ boards (like the Boards menu in the IDE). For a new user it's not clear that it only lists the boards _whose cores are installed_, also because `core install` is documented in the README after `board listall`.

```
$ arduino-cli board listall
Board Name FQBN

$ arduino-cli board
[...]
  listall     List all known boards and their corresponding FQBN.
[...]
```

### Suggested behavior

```
$ arduino-cli board listall
No known boards: install your first core with "arduino-cli core install <core>" to support its boards.
$ arduino-cli board
[...]
  listall     List all known boards (and their corresponding FQBN) according to the installed cores.
[...]
```

Also, would it be possible to add a `board search` command to list all the existing boards (even if not installed)? Do we have such data around?
Most users think like this: _I have a board_ -> _I know its name but I don't know what core it requires_ -> _Please install whatever core is needed for my board_

**Comment by alranel** [comment]
After running `arduino-cli core install arduino:samd`, the output of `arduino-cli board listall` is not empty. It was not clear to me that `board listall` depends on the installed cores (although it totally makes sense) because the README describes `core install` after `board listall`.
Maybe the description for `board listall` should mention that it lists the supported boards _according to the installed cores_ and that if you don't see your board listed you should install its core. I'd put this hint also in the output for `board listall` when no entries are returned.

The issue about `core search --additional-urls` is still there though.

**Comment by alranel** [comment]
On the other hand, would it be possible to list all the known boards even if their core is not installed?
Most users think like this: _I have a board_ -> _I know its name but I don't know what core it requires_ -> _Please install whatever core is needed for my board_

**Comment by alranel** [comment]
I opened a new issue for this in #550. The one about `core search --additional-urls` was partially fixed in #548
